packages/service-library/src/web-api/web-api-error-codes.ts

```diff
@@ -0,0 +1,106 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+// tslint:disable: no-unnecessary-class variable-name
+
+export declare type WebApiErrorCodeName =
+    | 'ResourceNotFound'
+    | 'InvalidResourceId'
+    | 'InvalidJsonDocument'
+    | 'RequestBodyTooLarge'
+    | 'InvalidURL'
+    | 'InternalError'
+    | 'MissingApiVersionQueryParameter'
+    | 'MissingContentTypeHeader'
+    | 'UnsupportedContentType'
+    | 'UnsupportedApiVersion';
+
```

Comment:
this change looks like it overlaps with the error codes feature. can you also address these if you are planning to cover that as part of this PR? 

1.) Invalid url
2.) Invalid page type (not a html page. Could be json/xml/image/etc url)
2.) Invalid input (missing required props, invalid json etc)
3.) Puppeteer crash
4.) timeout on loading
5) for other internal service issues like storage access error, queue access etc, we will have 1 error code - Internal Service Error
6.) For redirect url, we will not return error. Instead, the scan will be successful & return info about the redirected url with scan summary
7.) if failing for language issue, we need to see if we can identify this & have an error code for this. - so far, we have confirmed that we don't have language issues.
8.) axe scan error

Reply:
The current PR covers only HTTP responses w/o detecting backend failures